lib/operations/delete_many.js

```diff
@@ -18,8 +19,20 @@ class DeleteManyOperation extends OperationBase {
     const options = this.options;
 
     options.single = false;
-    removeDocuments(coll, filter, options, (err, r) => deleteCallback(err, r, callback));
+    removeDocuments(coll, filter, options, (err, r) => {
+      if (callback == null) return;
+      if (err && callback) return callback(err);
+      if (r == null) return callback(null, { result: { ok: 1 } });
+
+      // If an explain operation was executed, don't process the server results
+      if (this.explain) return callback(undefined, r.result);
+
```

Comment:
Got rid of `deleteCallback` and `updateCallback` in `common_functions` and moved the body of the functions into the respective operation classes; this way we can reference `this.explain` (and matches what happens in master).